distributed/http/tests/test_worker_http.py

```diff
@@ -35,3 +50,26 @@ def test_services(s, a, b):
     assert isinstance(c.services['http'], HTTPServer)
     assert c.service_ports['http'] == c.services['http'].port
     assert s.worker_services[c.address]['http'] == c.service_ports['http']
+
+
+@gen_cluster()
+def test_with_data(s, a, b):
+    e = Executor((s.ip, s.port), start=False)
+    yield tornado.gen.sleep(0.5)
+    yield e._start()
+    yield e._scatter([1])
```

Comment:
You'll want to capture the future here like this:

    future = yield e._scatter([1])

Otherwise, when the future gets garbage collected, the executor will send a signal to the scheduler that it is OK to delete the data you just scattered up.  Distributed memory mirrors local memory.